src/emulator/renderer/src/texture.cpp

```diff
@@ -82,5 +87,50 @@ void upload_bound_texture(const emu::SceGxmTexture &gxm_texture, const MemState
     glPixelStorei(GL_UNPACK_ROW_LENGTH, 0);
 }
 
+
+// Dumps bound texture to a file
+void dump(const emu::SceGxmTexture &gxm_texture, const MemState &mem, const std::string &base_path, const std::string &title_id) {
+    static uint32_t g_tex_index = 0;
+    static std::vector<uint8_t> g_pixels; // re-use the same vector instead of allocating one every time
+    static std::set<TextureCacheHash> g_dumped_hashes;
+
+    const TextureCacheHash hash = hash_texture_data(gxm_texture, mem);
+
+    if (g_dumped_hashes.find(hash) != g_dumped_hashes.end())
+        return;
+
+    g_dumped_hashes.insert(hash);
+
+    const size_t width = gxm::get_width(&gxm_texture);
+    const size_t height = gxm::get_height(&gxm_texture);
+    ;
```

Comment:
??

Reply:
i did this with no sleep man cut me some slack

Reply:
go to sleep!!!!

![image](https://user-images.githubusercontent.com/25717050/59198821-d085fd80-8bbe-11e9-82e0-23a8f38f78c9.png)

Reply:
lmao